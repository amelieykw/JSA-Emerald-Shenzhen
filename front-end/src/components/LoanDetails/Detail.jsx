import React, { useContext } from 'react'
import { Text, Grid, Col, Icon } from 'native-base'
import { TouchableOpacity, View } from 'react-native'
import { NavigationContext } from 'react-navigation'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import style from './DetailStyle'

const Detail = ({ loanList, id }) => {
  const navigation = useContext(NavigationContext)
  const { type, remaining } = loanList.find(loan => loan.id === id)

  const handlePress = () => {
    navigation.navigate('PayNow')
  }

  return (
    <Grid style={style.body}>
      <Col style={style.col}>
        <Text style={style.id}>{`ID - ${id}`}</Text>
        <Text style={style.name}>{type.name}</Text>
        <Text>{`Current Interest rate: ${type.interest}%`}</Text>
      </Col>
      <Col style={style.col}>
        <View>
          <Text style={style.remaining}>{`$ ${remaining}`}</Text>
          <Text style={style.remainingText}>remaining</Text>
        </View>
        <TouchableOpacity style={style.button} onPress={handlePress}>
          <View style={style.circle}>
            <Icon type="FontAwesome" name="dollar" style={style.dollar} />
          </View>
          <Text>PAY NOW</Text>
        </TouchableOpacity>
      </Col>
    </Grid>
  )
}

const mapStateToProps = state => {
  return { loanList: state.loanList, id: state.detailLoanId }
}

Detail.propTypes = {
  loanList: PropTypes.arrayOf(PropTypes.any).isRequired,
  id: PropTypes.number.isRequired
}

export default connect(mapStateToProps, null)(Detail)
