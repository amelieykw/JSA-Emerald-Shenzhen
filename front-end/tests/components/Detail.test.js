import React from 'react'
import thunk from 'redux-thunk'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'

import data from '../../helpers/mockData_FE'
import Detail from '../../src/components/LoanDetails/Detail'

const mockStore = configureStore([thunk])

describe('<Detail />', () => {
  let store
  let component
  beforeEach(() => {
    store = mockStore({ loanList: data.detailedLoans, detailLoanId: 1, loading: true })
  })

  it('<LoanDetails /> should render correctly', () => {
    component = renderer.create(
      <Provider store={store}>
        <Detail />
      </Provider>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
