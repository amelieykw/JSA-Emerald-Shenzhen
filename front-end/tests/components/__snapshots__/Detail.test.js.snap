// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Detail /> <LoanDetails /> should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#E3F2FD",
      "flex": 1,
      "flexDirection": "row",
      "height": 120,
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "color": "#7A7A7A",
            "fontSize": 18,
          },
        ]
      }
      uppercase={false}
    >
      ID - 1
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "color": "#5A5C60",
            "fontSize": 32,
          },
        ]
      }
      uppercase={false}
    >
      Student Loan
    </Text>
    <Text
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
      uppercase={false}
    >
      Current Interest rate: 0.03%
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
      }
    }
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "alignSelf": "flex-end",
              "color": "#FC815E",
              "fontSize": 36,
            },
          ]
        }
        uppercase={false}
      >
        $ 3400
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "alignSelf": "flex-end",
              "color": "#7A7A7A",
              "fontSize": 12,
              "paddingRight": 5,
            },
          ]
        }
        uppercase={false}
      >
        remaining
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "black",
            "borderRadius": 20,
            "height": 21,
            "marginRight": 8,
            "width": 21,
          }
        }
      >
        <Text />
      </View>
      <Text
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          }
        }
        uppercase={false}
      >
        PAY NOW
      </Text>
    </View>
  </View>
</View>
`;
